gdal/frmts/netcdf/netcdfsgwriterutil.cpp

```diff
@@ -208,6 +208,21 @@ namespace nccfdriver
         this->geometry_ref = ft.GetGeometryRef();
     }
 
+    inline void WBuffer::addCount(unsigned long long memuse)
+    {
+        this->used_mem += memuse;
+    }
+
+    inline void WBuffer::subCount(unsigned long long memfree)
+    {
+        unsigned long long oldmem = this->used_mem;
```

Comment:
compilers in non debug mode complain about oldmem variable not being used, since CPLAssert() is a no-op then. Could you rewrite this just as at it, that is
```
CPLAssert(this->used_mem >= memfree);
this->used_mem -= memfree;
```

Reply:
Yeah, your way just seems "better" in general, will fix it :+1: